pylintrc

```diff
@@ -184,7 +184,7 @@ evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / stateme
 [BASIC]
 
 # Good variable names which should always be accepted, separated by a comma
-good-names=main,_
+good-names=main,_,G
```

Comment:
G is added as a variable name for a generic graph as I understand? Could we use e.g.  `graph` instead? 

FYI on why:
One-letter function/variable names violate most styleguides (see e.g. https://www.python.org/dev/peps/pep-0008/#function-and-variable-names), especially if they are upper case. 

The rationale is normally the following ([here](https://github.com/kettanaito/naming-cheatsheet)'s a cool naming cheat-sheet, if you're interested):
One-letter variables are hard to interpret by users/readers (without context - they tend to require the user to scroll to the definition of the variable to see what its meaning is). For simple scripts like ours that's easy to do, for huge projects that can become a problem. The rationale behind the no-upper-case is that complete upper case names are reserved for constants (so a user knows they should never change this) or class definitions (e.g. Graph or DataFrame). If people strictly follow this rule then you always know what you're dealing with (a variable/function, a constant or a class) without having to check back. Ideally, variable names should be so intuitive that it would be the one of the first things you would guess.

Reply:
Yeah I agree. I was sticking with the standard notation of networkx library (e.g. https://networkx.org/documentation/stable/tutorial.html) but I don't actually like this either. I think they stuck with it because G the standard maths notation for a graph. But it's python and not maths, so very happy to switch to the more readable `graph`.

@herbiebradley: you might interested to also weigh in on this as well. I think you also used the G notation, would you be happy to switch to `graph` or would you rather stick with the networkx notation?

Reply:
I used G because it seems like one of the few cases where one letter variable names are intuitive and readable, but happy to switch to graph if we want.